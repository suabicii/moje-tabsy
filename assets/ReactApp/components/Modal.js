import React, {useEffect, useState} from "react";
import ReactModal from "react-modal";

function Modal(props) {
    const [showModal, setShowModal] = useState(false);
    const customStyles = {
        content: {
            top: '50%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            transform: 'translate(-50%, -50%)',
        },
    };

    useEffect(() => {
        setShowModal(props.modalIsOpen);
    });

    return (
        <ReactModal
            isOpen={showModal}
            appElement={document.getElementById('react')}
            shouldCloseOnOverlayClick={true}
            shouldCloseOnEsc={true}
            style={customStyles}
            contentLabel="Edycja"
        >
            {props.content}
        </ReactModal>
    );
}

export default Modal;