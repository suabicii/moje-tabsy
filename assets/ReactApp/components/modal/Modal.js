import React, {useEffect, useState} from "react";
import ReactModal from "react-modal";

function Modal({content, customRoot, modalIsOpen}) {
    const root = customRoot || document.getElementById('react'); // customRoot is for test purposes
    const [showModal, setShowModal] = useState(false);
    const customStyles = {
        content: {
            top: '50%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            transform: 'translate(-50%, -50%)'
        },
    };

    useEffect(() => {
        setShowModal(modalIsOpen);
    });

    ReactModal.defaultStyles.overlay.backgroundColor = 'rgba(0, 0, 0, 0.4)';

    return (
        <ReactModal
            isOpen={showModal}
            appElement={root}
            shouldCloseOnOverlayClick={true}
            shouldCloseOnEsc={true}
            style={customStyles}
            contentLabel="Edycja"
        >
            {content}
        </ReactModal>
    );
}

export default Modal;