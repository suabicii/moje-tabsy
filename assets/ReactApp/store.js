import {configureStore} from "@reduxjs/toolkit";
import drugsSlice from "./features/drugs/drugsSlice"
import userSlice from "./features/user/userSlice";
import darkModeSlice from "./features/darkMode/darkModeSlice";

export default configureStore({
    reducer: {
        drugs: drugsSlice,
        user: userSlice,
        darkMode: darkModeSlice
    }
});