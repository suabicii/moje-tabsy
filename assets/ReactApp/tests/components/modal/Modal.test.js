import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Modal from "../../../components/modal/Modal";
import {Provider} from "react-redux";
import store from "../../../store";

it('should correctly render opened modal with some content', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Provider store={store}>
            <Modal
                modalIsOpen={true}
                content={<p>Some content</p>}
                customRoot={<div className="custom-root"></div>}
            />
        </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});