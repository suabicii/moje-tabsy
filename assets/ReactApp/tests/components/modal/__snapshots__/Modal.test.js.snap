// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render opened modal with some content 1`] = `
<Modal
  appElement={false}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Edycja"
  isOpen={false}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": "auto",
        "left": "50%",
        "marginRight": "-50%",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      },
    }
  }
>
  <p>
    Some content
  </p>
</Modal>
`;
